src/expression.rs

```diff
@@ -142,27 +139,37 @@ impl ops::Add<Term> for Expression {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Iter<'a> {
-    iterator: slice::Iter<'a, Term>,
+    slice: &'a [Term],
 }
 
 impl<'a> Iterator for Iter<'a> {
     type Item = &'a Term;
 
     fn next(&mut self) -> Option<Self::Item> {
-        self.iterator.next()
+        self.slice.split_first().map(|(first, rest)| {
+            self.slice = rest;
+            first
+        })
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
 pub struct IterMut<'a> {
-    iterator: slice::IterMut<'a, Term>,
+    slice: &'a mut [Term],
 }
 
 impl<'a> Iterator for IterMut<'a> {
     type Item = &'a mut Term;
 
     fn next(&mut self) -> Option<Self::Item> {
-        self.iterator.next()
+        let slice = std::mem::take(&mut self.slice);
```

Comment:
workaround to satisfy borrow checker, as explained in [the Rustonomicon](https://doc.rust-lang.org/nomicon/borrow-splitting.html)